src/weights.jl

```diff
@@ -203,6 +219,13 @@ _wsum!(R::AbstractArray, A::AbstractArray, w::AbstractVector, dim::Int, init::Bo
 wsumtype{T,W}(::Type{T}, ::Type{W}) = typeof(zero(T) * zero(W) + zero(T) * zero(W))
 wsumtype{T<:BlasReal}(::Type{T}, ::Type{T}) = T
 
+
+"""
+    wsum!(R, A, w, dim; init=true)
```

Comment:
This:
```julia
init && fill!(R, zero(RT))
```

Apparently `init=false` won't start from zero when summing elements. So either you have already done this, or you want to add to the existing values. I guess we can just say: "if `init=false`, the sum is added to `R` instead of starting from zero".